plugins/InspectionGadgets/InspectionGadgetsAnalysis/src/com/siyeh/ig/performance/BoxingBoxedValueInspection.java

```diff
@@ -143,7 +159,7 @@ public void visitNewExpression(PsiNewExpression expression) {
       if (!constructorTypeText.equals(argumentTypeText)) {
         return;
       }
-      registerError(argument);
+      registerError(referenceNameElement, ProblemHighlightType.LIKE_UNUSED_SYMBOL, argument.getText());
```

Comment:
I am in the process of removing usages of LIKE_UNUSED_SYMBOL, now you should specify attributes in xml. Not sure if this inspection should be affected though

Reply:
Is it pushed, @[akozlova](https://github.com/akozlova)?

Reply:
I don't think the "Boxing of already boxed value" inspection should highlight as unused code. The code is not unused and that is in fact the problem with it, because it does unnecessary work.

Reply:
It's true that the code is actually used because it processes the data. The idea is to tell to the developer: _"If you remove this code, it will result to the same thing, so this code is useless."_ This way, to shorten the language and simplify the idea, we can think _as it_ does nothing. It is not actually true but it is as it was true.

Reply:
If the code should be coded another way, @[akozlova](https://github.com/akozlova), notify me and I will update the PR